### DISCUSSION TITLE
Multiple Queries share same data types

### DISCUSSION THREAD

**Discussion by reneschroeder0000** [ExpediaGroup/graphql-kotlin#1074] [Q&A] [answered]
Hi,

I would like to re-use fragments within multiple queries. Is there any possibility to do this? 
Use-case: findAll() and findById(id) endpoints shall use the same underlying data models. I want to prevent writing duplicated code to convert the returned data. Since I cannot create multiple queries in one document, I cannot use a fragment and therefore separate data classes are generated. 

Any help would be highly appreciated!

**Comment by dariuszkuc** [comment, marked as answer]
Hello 👋 
Unfortunately currently it is not possible to share any output types. With the latest changes (`4.0.0-alpha.15`) our generator was updated to allow sharing input objects, enums and custom scalars as they will always be the same across ALL operations. Output object types, interfaces and unions are always regenerated for each specified query/mutation as they can vary in their selection sets.

**Comment by ermadmi78** [comment]
Kobby plugin generates DSL client that provides an ability to reuse generated classes
https://github.com/ermadmi78/kobby
